### PR TITLE
Support native-json format log in kafka-fetcher-plugin

### LINKED ISSUE #6261: Support elastic / beats log-receiver
[Elastic filebeats](https://github.com/elastic/beats) is still released under Apache 2.0. So, we could consider to support its format as a new log receiver too.

@dmsolr leads on this.

### PR DESCRIPTION
**Pull request by hailin0**
### Support native-json format log in kafka-fetcher-plugin
- [ ] If this is non-trivial feature, paste the links/URLs to the design doc.
- [x] Update the documentation to include this new feature.
- [x] Tests(including UT, IT, E2E) are added to verify the new feature.
- [ ] If it's UI related, attach the screenshots below.

- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #6261, #6260.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

### CODE REVIEW

File: docs/en/setup/backend/log-analyzer.md [hunk 1]
```diff
@@ -11,8 +11,29 @@ Java agent provides toolkit for
...
+- [log4j2.x fileAppender](../../../../test/e2e/e2e-service-provider/src/main/resources/log4j2.xml)
+- [logback fileAppender](../../../../test/e2e/e2e-service-provider/src/main/resources/logback.xml)
+
+You can use [Filebeat](https://www.elastic.co/cn/beats/filebeat) 、[Fluentd](https://fluentd.org) to
+collect file logs including to use Kafka MQ to transport logs. When use this, need to open [kafka-fetcher](backend-fetcher.md#kafka-fetcher)
+and enable configs `enableNativeJsonLog`.
+
+Collector config examples:
+- [filebeat.yml](../../../../test/e2e/e2e-test/docker/kafka/filebeat.yml)
+- [fluentd.conf](../../../../test/e2e/e2e-test/docker/kafka/fluentd.conf)
```
Comment (wu-sheng) [inline docs/en/setup/backend/log-analyzer.md:36]: We need an update here, https://github.com/apache/skywalking/blob/master/docs/en/protocols/Log-Data-Protocol.md, to update log's JSON format. We don't implement HTTP JSON format receiver yet, right?

**Comment by wu-sheng** [comment]
Seems Kafka e2e fails, please recheck. cc @kezhenxu94

File: oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/provider/KafkaFetcherProvider.java [hunk 2]
```diff
@@ -77,9 +78,13 @@ public void start() throws ServiceNotProvidedException {
         if (config.isEnableMeterSystem()) {
             handlerRegister.register(new MeterServiceHandler(getManager(), config));
         }
-        if (config.isEnableLog()) {
+        if (config.isEnableNativeProtoLog()) {
             handlerRegister.register(new LogHandler(getManager(), config));
         }
+        if (config.isEnableNativeJsonLog()) {
+            handlerRegister.register(new JonLogHandler(getManager(), config));
```
Comment (kezhenxu94) [inline oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/provider/KafkaFetcherProvider.java:85]:
```suggestion
            handlerRegister.register(new JsonLogHandler(getManager(), config));
```

File: oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/provider/handler/JonLogHandler.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.skywalking.apm.network.logging.v3.LogData;
+import org.apache.skywalking.oap.server.analyzer.agent.kafka.module.KafkaFetcherConfig;
+import org.apache.skywalking.oap.server.library.module.ModuleManager;
+
+@Slf4j
+public class JonLogHandler extends LogHandler {
+
+    private static final JsonFormat.Parser JSON_PARSER = JsonFormat.parser().ignoringUnknownFields();
```
Comment (kezhenxu94) [inline oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/provider/handler/JonLogHandler.java:32]: Can you reuse https://github.com/apache/skywalking/blob/d56b0b0d41eefe4b81c02aa8f4c4b9c75721acf4/oap-server/server-library/library-util/src/main/java/org/apache/skywalking/oap/server/library/util/ProtoBufJsonUtils.java#L47

File: docs/en/setup/backend/log-analyzer.md (hunk 1 above)
Comment (hailin0) [reply docs/en/setup/backend/log-analyzer.md:36]: `We don't implement HTTP JSON format receiver yet, right?`

Yes, need to add and merge into this branch?

Comment (wu-sheng) [reply docs/en/setup/backend/log-analyzer.md:37]: We don't need to. I mean we should update the doc.

File: oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/provider/handler/JsonLogHandler.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.skywalking.apm.network.logging.v3.LogData;
+import org.apache.skywalking.oap.server.analyzer.agent.kafka.module.KafkaFetcherConfig;
+import org.apache.skywalking.oap.server.library.module.ModuleManager;
+import org.apache.skywalking.oap.server.library.util.ProtoBufJsonUtils;
+
+@Slf4j
+public class JsonLogHandler extends LogHandler {
+
+    private static final JsonFormat.Parser JSON_PARSER = JsonFormat.parser().ignoringUnknownFields();
```
Comment (kezhenxu94) [inline oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/provider/handler/JsonLogHandler.java:33]: Unused

File: oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/provider/handler/JsonLogHandler.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+    private final KafkaFetcherConfig config;
+
+    public JsonLogHandler(ModuleManager moduleManager, KafkaFetcherConfig config) {
+        super(moduleManager, config);
+        this.config = config;
```
Comment (kezhenxu94) [inline oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/provider/handler/JsonLogHandler.java:39]: You may also want to override the fields `histogram` and `errorCounter` to distinguish metrics for `native-proto` and `native-json`?

**Review by kezhenxu94 — COMMENTED** [review]
The tests and codes looks generally good to me, please fix the test.

**Review by wu-sheng — APPROVED** [review]
LGTM. @kezhenxu94 Please recheck the details.

**Review by kezhenxu94 — APPROVED** [review]
LGTM

**Comment by wu-sheng** [comment]
GHA crashes again. We have to wait for it back, then recheck all CI tasks.

**Comment by kezhenxu94** [comment]
> GHA crashes again. We have to wait for it back, then recheck all CI tasks.

The Kafka log task was run before GHA crashed, the logs shows there is still container failed to start up

**Review by wu-sheng — CHANGES_REQUESTED** [review]
Please fix the tests. It still fails.

**Review by dmsolr — DISMISSED** [review]
lgtm

**Review by kezhenxu94 — APPROVED** [review]

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**